### DISCUSSION TITLE
Strange artifacts in some plots in python binding

### DISCUSSION THREAD

**Discussion by hinxx** [epezent/implot#221] [General]
I know this might not be an ImPlot issue, but I hope to get some insight on where to look for solutions.

I'm trying to create a python binding for the ImPlot. I'm building on top of pyimgui concepts. So far I have made some good progress but I am experiencing some strange artifacts in some of the plots.
In the video below I'm running both ImGui plots and ImPlot plots; both from demo widgets.

https://user-images.githubusercontent.com/2725009/115846609-5fead180-a422-11eb-83a1-cd99e1abc7b2.mp4

One can notice the strange plotting on the left hand side (ImPlot plots), while the right hand side (ImGui plot) looks OK. This is just to put some context around what works for me at the moment.

Couple of other things I noticed. The Scatter plot looks OK, if I zoom in/out there is no artifacts seen. Bar plot also looks OK (can not zoom there).
Other ploy types, on the other hand, like Infinite lines, Stem plots, Error bars, Shadded and others do experience the similar problems as seen in the video.

FWIW, I'm doing this on the Linux OS, glfw3 and opengl. I use `#define ImDrawIdx unsigned int` in the `imconfig.h`.

Any insights on what is going on really appreciated!

**Comment by hinxx** [comment]
I noticed that if I enable the 'Anti-Aliased Lines' for the plot, the problem goes away. Here a demo of that.


https://user-images.githubusercontent.com/2725009/116515247-4d621380-a8cc-11eb-8335-ffe8c7f6bcbf.mp4


This seems to work for Line Plots, but only partly for Shaded Plots. In other words, it is not a complete solution.
Does this suggest that problem is in the render or backend?

**Reply by hinxx** [reply]
Solved it.
`ImDrawIdx` was set to `unsigned int` for ImGui compilation, but not for ImPlot :facepalm:.
Things look normal now!
Sorry for the noise!

**Reply by epezent** [reply]
No worries and apologies for the slow response! Glad you discovered the compilation issue, as I wouldn't have had any other suggestions than to reinvestigate 32-bit indices.
